File: Rust/Hello/hello.rs
use std::io;

fn read<T: std::str::FromStr>() -> T {
  let mut s = String::new();
  io::stdin().read_line(&mut s).ok();
  s.trim().parse().ok().unwrap()
}

fn main() {
  let name: String = read();
  println!("Hello, {}", name);
}

File: Erlang/Hello/hello.erl
-module(hello).
-export([main/0]).

main() ->
  Name = read_string(),
  io:format("Hello, ~s~n", [Name]).

read_string() ->
  case io:fread("", "~s") of
    eof -> 0;
    {ok, X} ->
      [Y] = X,
      Y
  end.
